parseddocument.cpp

```diff
@@ -23,63 +23,75 @@
 **
 ****************************************************************************/
 
-#include <memory>
-
 #include "parseddocument.h"
 
+#include <memory>
+
 using namespace Asn1Acn::Internal;
 
 ParsedDocument::ParsedDocument() :
     m_filePath(QString()),
-    m_revision(-1)
+    m_revision(-1),
+    m_parsedData(nullptr)
 {
 }
 
 ParsedDocument::ParsedDocument(const QString &filePath, int revision, const QStringList &wordList) :
     m_filePath(filePath),
-    m_revision(revision)
+    m_revision(revision),
+    m_wordList(wordList),
+    m_parsedData(nullptr)
 {
-    m_parsedTreeRoot = std::shared_ptr<ParsedObject>(new ParsedObject(m_filePath));
-
-    foreach (const QString &entry, wordList)
-        m_parsedTreeRoot->addChild(std::shared_ptr<ParsedObject>(new ParsedObject(QVariant(entry))));
 }
 
 ParsedDocument::ParsedDocument(const QString &filePath,
                                int revision,
                                std::unique_ptr<Data::Modules> parsedData) :
     m_filePath(filePath),
-    m_revision(revision)
+    m_revision(revision),
+    m_parsedData(std::move(parsedData))
+{
+}
+
+int ParsedDocument::getRevision() const
 {
-    m_parsedTreeRoot = std::shared_ptr<ParsedObject>(new ParsedObject(m_filePath));
+    return m_revision;
+}
 
+void ParsedDocument::bindParsedTreeNode(ParsedTreeNode::ParsedTreeNodePtr node)
+{
+    if (m_parsedData != nullptr) {
+        bindParsedTreeNodeWithParsedData(node);
+    } else if (m_wordList.empty() == false) {
+        bindParsedTreeNodeWithStubbedData(node);
+    }
```

Comment:
for QtCreator's Coding Rules:
> Use curly braces when the body of a conditional statement contains more than one line, and also if a single line statement is somewhat complex. Otherwise, omit them: